src/cuga/backend/server/demo_manage_setup.py

```
@@ -679,67 +681,29 @@ def seed_demo_knowledge_oobe_pdf_if_needed(demo_port: int, agent_id: str = "cuga
-                logger.warning("OOBE PDF ingest failed (%s): %s", r2.status_code, r2.text[:500])
-            else:
-                logger.info("Ingested OOBE knowledge PDF %s", DEMO_KNOWLEDGE_OOBE_PDF_NAME)
+        docs = await engine.list_documents(collection)
+        if any(d.filename == DEMO_KNOWLEDGE_OOBE_PDF_NAME for d in docs):
+            logger.info("OOBE knowledge PDF already indexed; skip ingest")
+            return
+        await engine.ingest(collection, pdf_path, True, DEMO_KNOWLEDGE_OOBE_PDF_NAME)
+        logger.info("Ingested OOBE knowledge PDF %s", DEMO_KNOWLEDGE_OOBE_PDF_NAME)
+    except DocumentExistsError:
+        logger.info("OOBE knowledge PDF already present; skip ingest")
     except Exception as e:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)